app/controllers/application_controller.rb

```diff
@@ -1,2 +1,9 @@
+# frozen_string_literal: true
+
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
+
+  def default_url_options
```

Comment:
Refactor(required): Porque você acha que isso é necessário? Isso é utilizado pelo devise para recursos de rota.

Só que esses recursos são usados além dos controllers, mailers utilizam também (quando o devise envia email para recuperar senha por exemplo). Por isso, o local dessa configuração está impróprio. Deve ser configurado a nível de enviroment (ou seja, pode ser diferente em desenvolvimento, staging, produção).

Vou deixar o local correto pra vc pensar a respeito. E desculpe, `request.host_with_port` é CAGADO, me falta portugues pra adjetivar melhor isso. Pra não dizer que sou ruim, segue uma doc [aqui](https://guides.rubyonrails.org/action_mailer_basics.html#generating-urls-in-action-mailer-views)